CMakeLists.txt

```diff
@@ -43,10 +43,18 @@ list(GET version_numbers 0 VERSION_MAJOR)
 list(GET version_numbers 1 VERSION_MINOR)
 set(VERSION_MAJ_MIN "${VERSION_MAJOR}.${VERSION_MINOR}")
 
-set(default_log_level "4")
-if (CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL RelWithDebInfo)
-  set(default_log_level "2")
+# make sure log level is defined and all-uppercase
+if (NOT VAST_LOG_LEVEL)
+  # use INFO as default level for release builds and TRACE otherwise
+  if (CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL RelWithDebInfo)
+    set(VAST_LOG_LEVEL "INFO")
+  else ()
+    set(VAST_LOG_LEVEL "TRACE")
+  endif ()
+else ()
+  string(TOUPPER "${VAST_LOG_LEVEL}" VAST_LOG_LEVEL)
```

Comment:
This is not quite right, VAST_LOG_LEVEL ought to be a CACHE variable, otherwise it has to be set to the requested value of every subsequent run of cmake in an existing build folder.
I understand that one might want to be able to change the log level of an existing build (which should work, because config.hpp should be regenerated). For that to work, we could compare the user provided value with the value in the cache like so:

``` cmake
set(default_log_level "TRACE")
if (CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL RelWithDebInfo)
  set(default_log_level "INFO")
endif ()
set(VAST_LOG_LEVEL_DOCSTRING
    "maximum log level that can be set during runtime execution")
if ("${VAST_LOG_LEVEL}" AND NOT "${VAST_LOG_LEVEL}" STREQUAL "$CACHE{VAST_LOG_LEVEL}")
  string(TOUPPER "${VAST_LOG_LEVEL}" VAST_LOG_LEVEL)
  set(VAST_LOG_LEVEL ${VAST_LOG_LEVEL} CACHE STRING
      "${VAST_LOG_LEVEL_DOCSTRING}" FORCE)
endif ()
set(VAST_LOG_LEVEL ${default_log_level} CACHE STRING
    "${VAST_LOG_LEVEL_DOCSTRING}")
```

Also, you forgot to remove L58-59, so as is, if VAST_LOG_LEVEL is not set from the command line, the default would get overridden by an string there.

Reply:
That's a lot of extra complexity only to get this thing cached. Franky, I would rather drop the `CACHE` line altogether for keeping it simple. I don't see it providing any value.

Reply:
The complexity is not for making the variable cached, but for allowing it to change in subsequent runs. Having a normal cached variable is even simpler than the code in this PR, as it was before.

Reply:
Ok, makes sense. Fixed.